applications/nrf_desktop/src/modules/Kconfig.dvfs

```diff
@@ -41,24 +40,35 @@ config DESKTOP_DVFS_RETRY_COUNT
 	  Number of retries of DVFS frequency change after which DVFS module will report
 	  MODULE_STATE_ERROR.
 
-config DESKTOP_DVFS_SMP_TRANSFER_TIMEOUT_MS
-	int "SMP transfer timeout"
-	default 5000
-	range 1 65535
-	help
-	  Timeout in milliseconds specifying time after which DVFS module will go back to
-	  low frequency after receiving SMP transfer event.
-
-config DESKTOP_DVFS_CONFIG_CHANNEL_TIMEOUT_MS
-	int "Config channel timeout"
-	default 5000
-	range 1 65535
-	help
-	  Timeout in milliseconds specifying time after which DVFS module will go back to
-	  low frequency after reciving config event.
-
 module = DESKTOP_DVFS
 module-str = DVFS
 source "subsys/logging/Kconfig.template.log_config"
 
+# Cannot leave this empty, as it will cause Kconfiglib to fail so add always met dependency
+dvfs_dependencies = y
+dvfs_state = DVFS_STATE_INITIALIZING
+# We start with DVFS_STATE_INITIALIZING on as it is active on start.
+config DVFS_STATE_INITIALIZING_INITIAL_ACTIVE
+	default y
```

Comment:
I would add variables per every configuration that you want to modify from the template instead of mixing the template approach with the default/choice overrides. Additionally, I would prefix the variables in the unified way:

```suggestion
dvfs_state = INITIALIZING
dvfs_state_active_freq_default = HIGH
# No dependencies are required for this state.
dvfs_state_dependencies = y
dvfs_state_initial_active = y
```

Reply:
And then we will have to have variable for each state for every kconfig option. I don't like it.

Reply:
I guess I suggested same thing in the https://github.com/nrfconnect/sdk-nrf/pull/16899#discussion_r1728879032 and https://github.com/nrfconnect/sdk-nrf/pull/16899#discussion_r1728882631 so +1 for @kapi-no suggestion